src/Spaces/dss2.jl

```diff
@@ -0,0 +1,461 @@
+                        p∂x∂ξ[level, ip, jp, i12, elem] *
+                        pdata[level, ip, jp, fidx + 1, elem]
+                    @inbounds pperimeter_data[level, p, fidx + 1, elem] =
+                        p∂x∂ξ[level, ip, jp, i21, elem] *
+                        pdata[level, ip, jp, fidx, elem] +
+                        p∂x∂ξ[level, ip, jp, i22, elem] *
+                        pdata[level, ip, jp, fidx + 1, elem]
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+function dss_transform2!(
+    dss_buffer,
+    data,
+    local_geometry,
+    weight,
+    perimeter,
+    localelems,
+)
+    if !isempty(localelems)
+        (; scalarfidx, vectorfidx, perimeter_data) = dss_buffer
+        ∂ξ∂x = local_geometry.∂ξ∂x
+        dss_transform2!(
+            perimeter_data,
+            data,
+            ∂ξ∂x,
+            weight,
+            perimeter,
+            scalarfidx,
+            vectorfidx,
+            localelems,
+        )
+    end
+    return nothing
+end
+
+function dss_untransform2!(dss_buffer, data, local_geometry, perimeter)
+    (; vectorfidx, perimeter_data) = dss_buffer
+    ∂x∂ξ = local_geometry.∂x∂ξ
+    dss_untransform2!(perimeter_data, data, ∂x∂ξ, perimeter, vectorfidx)
+    return nothing
+end
+
+function dss_local2!(perimeter_data, perimeter, topology)
+    (Np, _, _, Nv, _) = size(perimeter_data)
+    nfacedof = div(Np - 4, 4)
+
+    @inbounds for vertex in Topologies.local_vertices(topology)
+        # for each level
+        @inbounds for level in 1:Nv
+            # gather: compute sum over shared vertices
+            sum_data = mapreduce(⊞, vertex) do (lidx, vert)
+                ip = Topologies.perimeter_vertex_node_index(vert)
+                perimeter_slab = slab(perimeter_data, level, lidx)
+                perimeter_slab[ip]
+            end
+            # scatter: assign sum to shared vertices
+            @inbounds for (lidx, vert) in vertex
+                perimeter_slab = slab(perimeter_data, level, lidx)
+                ip = Topologies.perimeter_vertex_node_index(vert)
+                perimeter_slab[ip] = sum_data
+            end
+        end
+    end
+
+    @inbounds for (lidx1, face1, lidx2, face2, reversed) in
+                  Topologies.interior_faces(topology)
+        pr1 = Topologies.perimeter_face_indices(face1, nfacedof, false)
+        pr2 = Topologies.perimeter_face_indices(face2, nfacedof, reversed)
+        @inbounds for level in 1:Nv
+            perimeter_slab1 = slab(perimeter_data, level, lidx1)
+            perimeter_slab2 = slab(perimeter_data, level, lidx2)
+            @inbounds for (ip1, ip2) in zip(pr1, pr2)
+                val = perimeter_slab1[ip1] ⊞ perimeter_slab2[ip2]
+                perimeter_slab1[ip1] = val
+                perimeter_slab2[ip2] = val
+            end
+        end
+    end
+    return nothing
+end
+
+function dss_local_ghost2!(perimeter_data, perimeter, topology)
```

Comment:
does "local" part of ghost vertex dss (e.g. where 2 local elements share a vertex that needs to be computed). This is stored back in all locations in `perimeter_data`